File: code_snippets/ts/src/develop/error_handling.ts
import { TerminalError } from "@restatedev/restate-sdk";
import * as restate from "@restatedev/restate-sdk";

const service = restate.service({
  name: "ErrorHandling",
  handlers: {
    greet: async (ctx: restate.Context, name: string) => {
      // <start_here>
      throw new TerminalError("Something went wrong.", { errorCode: 500 });
      // <end_here>
    },
  },
});

File: code_snippets/go/develop/errorhandling.go
package develop

import (
	"fmt"

	restate "github.com/restatedev/sdk-go"
)

type ErrorHanding struct{}

func (ErrorHanding) Greet(ctx restate.Context, name string) error {
	// <start_here>
	return restate.TerminalError(fmt.Errorf("Something went wrong."), 500)
	// <end_here>
}
